### PR TITLE
Internal refactoring

### PR DESCRIPTION
**Pull request by pirelenito**
Improves internal code organization, making it simpler for external contributors to extend and fix Sagui.

This is a long overdue change!

The diff can be a little overwhelming, so it might be easier to take a look at https://github.com/saguijs/sagui/tree/internal-refactoring/src

### CODE REVIEW

**Comment by pirelenito** [comment]
I want to come back to Webpack 2 migration right after this!

**Comment by pirelenito** [comment]
Just release a RC, to make it easier to try it out on your projects.

https://github.com/saguijs/sagui/releases/tag/v8.0.5-rc.0

Let me know how it goes!

Thanks

File: src/index.js [hunk 1]
```diff
@@ -1,96 +1,56 @@
...
+ * @param {string[]} [saguiConfig.javaScript.buildDependencies = true] Which dependencies to transpile (Ex: ['ui-react-components'])
+ * @param {boolean} [saguiConfig.optimize = false] Optimize the output (minify, dedup...)
+ * @param {boolean} [saguiConfig.coverage = false] Outputs test coverage while running the tests
+ * @param {string[]} [saguiConfig.pages = []] Define a build output based on a HTML and JS files.
+ * @param {string[]} [saguiConfig.disabledLoaders = []] Disables loaders for specific file types.
+ * @param {Object} [saguiConfig.webpack] Webpack configuration object to extend the internal configuration.
+ * @param {Object} [saguiConfig.karma] Karma configuration object to extend the internal configuration.
+ */
+const sagui = (saguiConfig = {}) => {
+  const finalSaguiConfig = {
```
Comment (xaviervia) [inline src/index.js]: 👏

File: src/load-project-sagui-config.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+function sanityCheck (projectPath) {
+  const packagePath = path.join(projectPath, 'package.json')
+  if (!fileExists(packagePath)) throw new MissingPackageJSON()
+
+  const packageJSON = json.read(packagePath)
+  if (packageJSON.name === 'sagui') throw new SaguiPath()
+}
+
+export function MissingPackageJSON () {
```
Comment (xaviervia) [inline src/load-project-sagui-config.js]: Fancy!

**Comment by xaviervia** [comment]
Great job! Amazing. Looks far easier to follow than before.

File: src/load-project-sagui-config.js (hunk 2 above)
Comment (pirelenito) [inline src/load-project-sagui-config.js]: To be clear, this was moved from `src/index.js`. 😀

**Pull request merged**